src/webviews/extension-side/dataviewer/interpreterDataViewerDependencyImplementation.node.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { SemVer } from 'semver';
+import { CancellationToken, CancellationTokenSource } from 'vscode';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { IInstaller, Product, InstallerResponse } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { Cancellation, createPromiseFromCancellation } from '../../../platform/common/cancellation';
+import { traceWarning } from '../../../platform/logging';
+import { IPythonExecutionFactory } from '../../../platform/common/process/types.node';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { IInterpreterService } from '../../../platform/interpreter/contracts';
+import { PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { pandasMinimumVersionSupported } from './constants';
+
+/**
+ * Uses the Python interpreter to manage dependencies of a Data Viewer.
+ */
+export class InterpreterDataViewerDependencyImplementation implements IDataViewerDependencyService {
+    constructor(
+        private readonly applicationShell: IApplicationShell,
+        private readonly installer: IInstaller,
+        private pythonFactory: IPythonExecutionFactory,
+        private interpreterService: IInterpreterService,
+        private isCodeSpace: boolean
+    ) {}
+
+    public async checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
+        const tokenSource = new CancellationTokenSource();
+        try {
+            const pandasVersion = await this.getVersion(interpreter, tokenSource.token);
+
+            if (Cancellation.isCanceled(tokenSource.token)) {
+                return;
+            }
+
+            if (pandasVersion) {
+                if (pandasVersion.compare(pandasMinimumVersionSupported) > 0) {
+                    return;
+                }
+                sendTelemetryEvent(Telemetry.PandasTooOld);
+                // Warn user that we cannot start because pandas is too old.
+                const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
```

Comment:
Why not just use `pandasVersions.raw` or the like, instead of formatting this manually.

Reply:
The reason why this isn't done in the interpreter version is because `pandasVersions.raw` results in `0.20..` 🤔